python/cudf/cudf/dataframe/series.py

```diff
@@ -115,6 +115,29 @@ def __init__(
     def from_pandas(cls, s, nan_as_null=True):
         return cls(s, nan_as_null=nan_as_null)
 
+    @property
+    def values(self):
+        def as_gpu_matrix(ser, columns=None, order="F"):
+            nrow = len(ser)
+            dtype = ser.dtype
+            if order == "F":
+                matrix = rmm.device_array(
+                    shape=(nrow,), dtype=dtype, order=order
+                )
+                dense = ser._column.to_gpu_array(fillna="pandas")
+                matrix[:].copy_to_device(dense)
+            elif order == "C":
+                matrix = cudautils.row_matrix([ser._column], nrow, 1, dtype)
+            else:
+                errmsg = (
+                    "order parameter should be 'C' for row major or 'F' for"
+                    "column major GPU matrix"
+                )
+                raise ValueError(errmsg.format(ser._column))
+            return matrix
+
+        return as_gpu_matrix(self).copy_to_host()
+
```

Comment:
Can't `Series.values` just be something like this:

```
@values
def values(self):
    if self.dtype == 'object':
        return self.data.to_host()
    else:
        return self.data.mem.copy_to_host()
```

Reply:
Yes, that would work but in case of categoricals pandas returns `pandas.core.arrays.categorical.Categorical` object which has categories in it:
```python
pd.Series(['a','b'], dtype='category').values
[a, b]
Categories (2, object): [a, b]
```

In this case should we just return our categorical column reference?

Reply:
If we absolutely need to return a host object then just:

```
a._column.to_pandas().values
```

:P

Reply:
At a higher level though, do we want `Series.values` to return a host object, or something that lives on the device like a `DeviceNDArray`? @beckernick any thoughts?

Reply:
Purpose of introducing this (`values`) properties to series was for dask to be able to work in a generic way. So I think it has to be a copy of values in host. But pandas `.values` gives reference of data. Dask expects a numpy like or a cupy like array.
Any comments on this @beckernick @mrocklin @kkraus14 ?

Reply:
I agree with @shwina 's comment. Let's revisit this discussion in a separate PR

Reply:
FWIW, my personal preference here would be for this to return a device-backed array-like object, perhaps something like a CuPy array.  

I think that moving it to host also makes sense, but might be less useful overall.  If this did show up in a separate issue/PR I'd love to jump in there.

Reply:
Once the cuML packaging issues are resolved we can revisit adding it as a dependency and returning it via `.values` here. That being said, I have serious concerns about modifying the numpy arrays under Pandas objects in place to also modify the Pandas object and us not having the same ability to replicate that behavior. To a certain extent, I'd almost just want to commit to cuDF Series supporting numpy api things via `__array_function__` wherever possible.